pkg/tuned/controller.go

```diff
@@ -918,6 +942,42 @@ func getNodeName() string {
 	return name
 }
 
+func (c *Controller) getNodeForProfile(nodeName string) (*corev1.Node, error) {
+	node, err := c.kubeclient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Node %s: %v", nodeName, err)
+	}
+	return node, nil
+}
+
+func (c *Controller) updateNodeAnnotations(node *corev1.Node, annotations map[string]string) (err error) {
+	var (
+		change bool
+	)
+	node = node.DeepCopy() // never update the objects from cache
+
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = map[string]string{}
+	}
+
+	for k, v := range annotations {
+		change = change || node.ObjectMeta.Annotations[k] != v
+		node.ObjectMeta.Annotations[k] = v
+	}
+
+	if !change {
+		// No Node annotations changed, no need to update
+		return nil
+	}
+
+	_, err = c.kubeclient.CoreV1().Nodes().Update(context.TODO(), node, metav1.UpdateOptions{})
```

Comment:
I think I need a better explanation about the security of this part. Is the annotation protected somehow? Is the kubelet client limited to updating only its own Node object?

In other words, please document the high level architecture and the rationale (the why) better here or in the commit message so we will not have to reread all the code to learn how it was supposed to work in the future.

Reply:
Added a comment + the higher level architecture is already documented in the commit message.